consensus/src/pending_votes.rs

```diff
@@ -168,14 +172,17 @@ impl PendingVotes {
         if let Some((timeout, signature)) = vote.two_chain_timeout() {
             let partial_tc = self
                 .maybe_partial_2chain_tc
-                .get_or_insert_with(|| TwoChainTimeoutCertificate::new(timeout.clone()));
+                .get_or_insert_with(|| TwoChainTimeoutWithPartialSignatures::new(timeout.clone()));
             partial_tc.add(vote.author(), timeout.clone(), signature.clone());
             let tc_voting_power = match validator_verifier.check_voting_power(partial_tc.signers())
             {
                 Ok(_) => {
-                    return VoteReceptionResult::New2ChainTimeoutCertificate(Arc::new(
-                        partial_tc.clone(),
-                    ))
+                    return match partial_tc.aggregate_signatures(validator_verifier, true) {
```

Comment:
So, I'm confused here...

Even if you fail to aggregate the signatures, this could be because some of the `n` signatures are invalid, but you might still have `2f+1` of them (or whatever your threshold is) that are valid. If so, giving up here might not be _correct_, no? Instead, you need to identify the correct signature shares and aggregate those.

(Recall that, in the last PR, you left a comment about me writing a faster algorithm for identifying these correct signature shares.)

Or is your threshold exactly `n`? I assume, no.

Reply:
We already optimistically verify all signatures individually, so here, once we have `2f+1` partial sig, there is no reason the aggregation will fail. Once we have optimistic signature aggregation support - that's where your suggestion make sense.